spirv_msl.cpp

```diff
@@ -5750,22 +5750,44 @@ void CompilerMSL::emit_header()
 {
 	// This particular line can be overridden during compilation, so make it a flag and not a pragma line.
 	if (suppress_missing_prototypes)
-		statement("#pragma clang diagnostic ignored \"-Wmissing-prototypes\"");
+		add_pragma_line("#pragma clang diagnostic ignored \"-Wmissing-prototypes\"");
```

Comment:
Using add_pragma_line is somewhat questionable here. This will force a recompile, even before we have committed to emitting the header. This is wasteful, and I think this needs to fix add_pragma_line so that it doesn't force a recompile unless it's called after the header is committed. I think all uses of add_pragma happen before we emit the header, so maybe that path can simply be dropped.

Reply:
I'll look into cleaning that up.